### PR TITLE
Add `init?(argbHexString:)` to support the common ARGB format used in Android

### PR DESCRIPTION
**Pull request by yonat**
🚀

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 10.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/SwifterSwift/Shared/ColorExtensions.swift [hunk 1]
```diff
@@ -332,6 +332,30 @@ public extension Color {
...
+    /// SwifterSwift: Create Color from hexadecimal string in the format ARGB (alpha-red-green-blue).
+    ///
+    /// - Parameters:
+    ///   - argbHexString: hexadecimal string (examples: 7FEDE7F6, 0x7FEDE7F6, #7FEDE7F6, #f0ff, 0xFF0F, ..).
+    convenience init?(argbHexString: String) {
+        var string = argbHexString.replacingOccurrences(of: "0x", with: "").replacingOccurrences(of: "#", with: "")
+
+        if string.count <= 4 { // convert hex to long format if in short format
+            var str = ""
+            string.forEach { str.append(String(repeating: String($0), count: 2)) }
```
Comment (guykogus) [inline Sources/SwifterSwift/Shared/ColorExtensions.swift:344]: Prefer `for-in` over `forEach` whenever possible.

File: Sources/SwifterSwift/Shared/ColorExtensions.swift [hunk 2]
```diff
@@ -332,6 +332,30 @@ public extension Color {
...
+            string = str
+        }
+
+        guard let hexValue = Int(string, radix: 16) else { return nil }
+        let hasAlpha = string.count == 8
+
+        let alpha = hasAlpha ? (hexValue >> 24) & 0xFF : 0xFF
```
Comment (guykogus) [inline Sources/SwifterSwift/Shared/ColorExtensions.swift:351]:
```suggestion
        guard let hexValue = Int(string, radix: 16) else { return nil }

        let hasAlpha = string.count == 8
        let alpha = hasAlpha ? (hexValue >> 24) & 0xFF : 0xFF
```

**Review by guykogus — CHANGES_REQUESTED** [review]
Nice. Just a couple of small points.

**Comment by yonat** [comment]
Thanks @guykogus , fixed.

**Review by guykogus — APPROVED** [review]

**Pull request merged**